### PR TITLE
gls_FY

### PR DESCRIPTION
**Pull request by EnricaBelfiore**
GLS update of fission yields and their related covariance matrix
---

- Added private method `sandy.Fy._gls_parameters_update()` to update IFY with the GLS update technique for a given zam, energy, design sensitivity and extra information
- Added private method `sandy.Fy._gls_cov_update()` to update the prior IFY covariance matrix using the GLS technique
- Added `sandy.Fy.gls_update()` to perform the GLS update of fission yields and their related covariance matrix, according with
 [Generation of fission yield covariances to correct discrepancies in the nuclear data libraries](https://www.sciencedirect.com/science/article/pii/S0306454915005137?via%3Dihub)

### CODE REVIEW

**Comment by EnricaBelfiore** [comment]
The example tests are numerically correct but they will be changed in the next commit to have a physical meaning. 
I think that in the docstring we should specify that the indices/columns of `S`, `y_extra` and `Vy_extra` must be coherent.

**Review by luca-fiorito-11 — CHANGES_REQUESTED** [review]
Need notebook

File: sandy/fy.py [hunk 1]
```diff
@@ -621,109 +622,174 @@ def gls_cov_update(self, zam, e, Vy_extra=None,
...
+        index = Vx_prior.index
+        S_ = pd.DataFrame(S).reindex(columns=index).fillna(0)
+        y_extra_ = pd.Series(y_extra).reindex(index=S_.index).fillna(0).values
+        # Perform GLS:
         if Vy_extra is not None:
             Vy_extra_ = pd.DataFrame(Vy_extra)
-            S = S.reindex(index=Vy_extra_.index).fillna(0)
-            Vx_post = Vx_prior.gls_cov_update(S, Vy_extra_).data
+            S_ = S_.reindex(index=Vy_extra_.index).fillna(0)
+            y_extra_ = pd.Series(y_extra).reindex(index=S_.index).fillna(0).values
```
[the remark this answers is not in the document]
Comment (EnricaBelfiore) [reply sandy/fy.py]: I agree, y_extra and V_extra must be pd.Series and pd.DataFrame and the names of the indeces must be
consistent with the names of the indices of the sensitivity matrix (like in the example tests). I will add it in the docstring

File: sandy/fy.py [hunk 2]
```diff
@@ -621,109 +622,174 @@ def gls_cov_update(self, zam, e, Vy_extra=None,
...
-            .reindex(columns=Vx_prior.data.columns).fillna(0)
+        # Filter FY data:
+        fy_data = self.data.query(f"ZAM=={zam} & E=={e} & MT==454").set_index('ZAP')
+        mat = fy_data.MAT.iloc[0]
+        x_prior = fy_data.FY
+        std = fy_data.DFY
+        Vx_prior = sandy.CategoryCov.from_stdev(std).data
+        index = Vx_prior.index
+        S_ = pd.DataFrame(S).reindex(columns=index).fillna(0)
+        y_extra_ = pd.Series(y_extra).reindex(index=S_.index).fillna(0).values
```
[the remark this answers is not in the document]
Comment (EnricaBelfiore) [reply sandy/fy.py]: The inputs of `sandy._gls_parameters_update` should be np.ndarray

**Review by luca-fiorito-11 — APPROVED** [review]

**Pull request merged**